2019-04-25 12:36:36,458 ERROR: Exception on /index [GET] [in C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "D:\PyCharm\Flask\app\routes.py", line 53, in index
    prev_url=prev_url, opros=False)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\PyCharm\Flask\app\templates\index.html", line 32, in template
    {% endblock %}
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 268, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 223, in parse_if
    drop_needle=True)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 144, in parse_statement
    self.fail_unknown_tag(token.value, token.lineno)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 97, in fail_unknown_tag
    return self._fail_ut_eof(name, self._end_token_stack, lineno)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 90, in _fail_ut_eof
    self.fail(' '.join(message), lineno)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'endif'. The innermost block that needs to be closed is 'if'.
2019-04-25 12:37:15,983 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 12:39:44,871 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 12:39:47,875 ERROR: Exception on /index [GET] [in C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "D:\PyCharm\Flask\app\routes.py", line 53, in index
    prev_url=prev_url, opros=False)
  File "C:\ProgramData\Anaconda3\lib\site-packages\flask\templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\PyCharm\Flask\app\templates\index.html", line 18, in template
    {% endif % }
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 268, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\parser.py", line 888, in subparse
    self.stream.expect('block_end')
  File "C:\ProgramData\Anaconda3\lib\site-packages\jinja2\lexer.py", line 384, in expect
    self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: expected token 'end of statement block', got '%'
2019-04-25 12:40:40,441 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 12:41:57,906 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 12:42:13,747 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 12:42:28,059 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 13:31:23,331 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
2019-04-25 13:31:41,760 INFO: Microblog startup [in D:\PyCharm\Flask\app\errors.py:45]
